src/steadystateproblem.cpp

```diff
@@ -206,21 +207,16 @@ void SteadystateProblem::applyNewtonsMethod(ReturnData *rdata, Model *model,
             /* Check residuals vs tolerances */
             converged = wrms < RCONST(1.0);
 
-            if (converged) {
-                /* Ensure positivity of the found state */
-                for (ix = 0; ix < model->nx_solver; ix++) {
-                    if (x[ix] < 0.0) {
-                        x[ix] = 0.0;
-                        converged = FALSE;
-                    }
-                }
```

Comment:
What is the reason to kick out this part? Usually, we want positive states...
Sure, this part may screw up convergence, if we converged to a steady state with negative entries... But usually, we don't want to keep this steady state with negative entries anyway...

Reply:
Makes problems when you have a zero steady-state (or at least zero at one index). Then you will never converge due to floating-point operations. The number -1e-32 is so close to zero that you do not want to force it to be 0.0. We had such issues.

On the other hand, for strictly positive steady-states this part of the code was never invoked.

Reply:
I see your point... but:
This part of Amici is sometimes used to preequilibrate systems and start a simulation after prequilibration, i.e., sometimes, the Ewton-solver is used to find inital values of ODEs. Negative states can then cause integration failure, when a simulation is started from them... We also experienced cases like this...

So, we must find a solution to this... 
I don't yet really understand what goes wrong if we just set those `x` to `0.0`. Can you re-explain, please?

About strict positivity: At least to my understanding, this is not what one is looking for. It is important to be `>= 0.0`. Please correct me here, if I'm wrong. Maybe @FFroehlich knows more about these cases than I do...

How about checking whether we are in between of `-absolute_tolerance` and `0.0`, an just flip sign in that case?

Reply:
I see the issues. Maybe a reasonable solution would be to have `if (x[i] < -eps)` for some epsilon.

In our case, with this part of the code and zero steady-states, the Newton solver never converged. Always reached the maximum number of iterations. In the debugging, I noticed the solution was forced to be zero and in the next step, it again went to negative (-1e-32). The process repeated all the time.

Reply:
Ah, okay, now I see what's wrong:
You're totally right about taking out the `converged = FALSE;`, indeed!
One thing: Can we agree on setting stuff to 0.0, and then just rechecking for convergence?
If convergence persists, we are fine and keep `converged = TRUE;` and hence drop out of the loop. Only if we have no convergence, we set  `converged = FALSE;` after reapplying the positivity constraint?

Reply:
Understand. I will check it out. Could be enough indeed.

Reply:
Does your code example still work with this adaptation now? If yes, I'd do a review for merging.

Reply:
I wouldn't have done this change without running my test first. It works indeed.

Reply:
Okay, point for you. ;)